stdlib.sh

```diff
@@ -26,6 +25,92 @@ direnv_config_dir=${XDG_CONFIG_DIR:-$HOME/.config}/direnv
 # algorithm and so it won't be re-exported.
 export DIRENV_IN_ENVRC=1
 
+__store_env_strictness() {
+  set +o | grep 'pipefail\|nounset\|errexit' > "/tmp/env_strictness.$$.$1"
+}
+
+__restore_env_strictness() {
+  local file
+  file="/tmp/env_strictness.$$.$1"
```

Comment:
Some environments use TMPDIR to set the temp dir location.
```suggestion
  file="${TMPDIR:-/tmp}/env_strictness.$$.$1"
```